paasta_tools/cli/cmds/spark_run.py

```diff
@@ -619,6 +627,9 @@ def get_spark_env(
         spark_env["SPARK_DAEMON_CLASSPATH"] = "/opt/spark/extra_jars/*"
         spark_env["SPARK_NO_DAEMONIZE"] = "true"
 
+    if args.use_eks:
+        spark_env["KUBECONFIG"] = system_paasta_config.get_spark_kubeconfig()
```

Comment:
hmm Interesting that this env variable wasn't set before.
What would this be this used for? I believe this file `/etc/kubernetes/spark.conf` is yet to be deployed via puppet?

Reply:
Noticed in description: "we'll be using a kubeconfig with the standard EKS setup rather than giving spark a URL "
Slightly confused here: The major difference I notice here is providing a k8s_server_address for spark master configuration. Is this k8s_servcer_address internally somehow loaded from given KUGECONFIG?

Reply:
> What would this be this used for? I believe this file /etc/kubernetes/spark.conf is yet to be deployed via puppet?

We are using this file to authenticate to the kubernetes API server, as we are no longer using the `spark.kubernetes.authenticate.clientKeyFile`/`spark.kubernetes.authenticate.clientCertFile` parameters to authenticate.  It appears we were never setting the KUBECONFIG envvar or using the puppet-deployed KUBECONFIG for the current, non-EKS clusters at all 😮 

This new kubeconfig file is already deployed by puppet, though is currently configured for our test EKS cluster "paasta-test" -- when we are able to spin up our productionized clusters, we will need to change puppet to target the correct cluster

> The major difference I notice here is providing a k8s_server_address for spark master configuration. Is this k8s_servcer_address internally somehow loaded from given KUGECONFIG?

For some reason, unfortuantely spark will not read the k8s_server_address from the specified KUBECONFIG, and so we need spark-run to still specify this in k8s_server_address as well.

Reply:
we probably could have saved ourselves a bit of effort/code in the spark k8s migration by using the KUBECONFIG env var rather than hardcoding cert locations and whatnot - but otherwise, what jfong said!

Reply:
[Not a blocker, for understanding] Is the plan to also deploy this file also on batch and dev boxes? I am assuming this is used by spark driver (running on batch box) to authenticate.

Reply:
yup - it should already be deployed on all batch and devboxes :)